Sources/Hero.swift

```diff
@@ -52,7 +52,9 @@ public class Hero: HeroBaseController {
   public internal(set) var fromViewController: UIViewController?
   /// whether or not we are presenting the destination view controller
   public internal(set) var presenting = true
-
```

Comment:
Instead of having this `reverseMode` boolean. May I suggest that we add a new enum case in `HeroDefaultAnimationType`. Something like 
```swift
enum HeroDefaultAnimationType {
  func autoReverse(_ anim: HeroDefaultAnimationType) -> HeroDefaultAnimationType {
    return .selectBy(presenting: anim, dismissing: anim.reverse())
  }
}
```

then use it like this:
```swift
heroModalAnimationType = .autoReverse(.push(direction:.left))
```